dependencies.yaml

```diff
@@ -710,36 +708,6 @@ dependencies:
           - *tensordict
           - pyg>=2.5,<2.6
 
-  depends_on_pytorch:
-    common:
-      - output_types: [conda]
-        packages:
-          - &pytorch_unsuffixed pytorch>=2.0,<2.2.0a0
-          - torchdata
-          - pydantic
-
-    specific:
-      - output_types: [requirements]
-        matrices:
-          - matrix: {cuda: "12.*"}
-            packages:
-              - --extra-index-url=https://download.pytorch.org/whl/cu121
-          - matrix: {cuda: "11.*"}
-            packages:
-              - --extra-index-url=https://download.pytorch.org/whl/cu118
-          - {matrix: null, packages: null}
-      - output_types: [requirements, pyproject]
-        matrices:
-          - matrix: {cuda: "12.*"}
-            packages:
-              - &pytorch_pip torch>=2.0,<2.2.0a0
-              - *tensordict
-          - matrix: {cuda: "11.*"}
-            packages:
-              - *pytorch_pip
-              - *tensordict
-          - {matrix: null, packages: [*pytorch_pip, *tensordict]}
-
```

Comment:
Think it is ok to keep this. We just want to relax the upper bound. At a minimum we need PyTorch 2.3.0 for NumPy 2 support ( https://github.com/pytorch/pytorch/issues/107302 ). Though wonder if this upper bound can be dropped altogether

Reply:
As long as we have cupy>13.2 in our test environment we are ok to remove the upper bound

Reply:
I'm also ok if we enforce pytorch >=2.3 for both test and runtime

Reply:
Let's keep the discussion about what to do with the `pytorch` dependency in one thread, for ease of linking.

@hcho3 would `>=2.3.0`, as @alexbarghi-nv suggested here, work? A floor would be preferable to a ceiling, I think...otherwise this will have to be changed again in the next release.

Note: we should match whatever's done here in https://github.com/rapidsai/wholegraph/pull/218

Reply:
No, because 2.4.0 has issues with Conda-forge packaging. See https://github.com/conda-forge/pytorch-cpu-feedstock/issues/254#issuecomment-2319090902

Reply:
This was recently resolved. So have submitted PR ( https://github.com/rapidsai/cugraph/pull/4703 ) to relax this in 24.12